airflow/models/taskinstance.py

```diff
@@ -197,7 +199,7 @@ def clear_task_instances(
             if dag and dag.has_task(task_id):
                 task = dag.get_task(task_id)
                 ti.refresh_from_task(task)
-                task_retries = task.retries
+                task_retries = task.retries or 0
```

Comment:
I also had to add this, as `task.retries` is `Optional[int]`. Not sure it would have worked if it was None, but I'll look closer tomorrow.

Reply:
It wouldn’t work, and honestly I don’t even know what `None` means here. I think this is exposing a more fundamental bug…